runners/flink/1.10/job-server/build.gradle

```diff
@@ -29,3 +29,7 @@ project.ext {
 
 // Load the main build script which contains all build logic.
 apply from: "$basePath/flink_job_server.gradle"
+
+dependencies {
+  permitUnusedDeclared project(":runners:flink:1.10")
```

Comment:
This seems strange. The Flink 1.10 job server should definitely actually use `:runners:flink:1.10`. And same for the other versions. Why is this needed?

Reply:
project(":runners:flink:1.10") is transitively being declared, though we do not need it here. If I dont add 

dependencies {
  permitUnusedDeclared project(":runners:flink:1.10") 
}

I have been getting,
What went wrong:
Execution failed for task ':runners:flink:1.10:job-server:analyzeClassesDependencies'.
> Dependency analysis found issues.
  unusedDeclaredArtifacts: 
   - beam.runners.flink:1.10:2.27.0-SNAPSHOT@jar


same is true for other versions.

Reply:
`:runners:flink:1.10:job-server:analyzeClassesDependencies `task fails if we don't add `permitUnusedDeclared project(":runners:flink:1.10")` 

```
Task :runners:flink:1.10:job-server:analyzeClassesDependencies FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':runners:flink:1.10:job-server:analyzeClassesDependencies'.
> Dependency analysis found issues.
  unusedDeclaredArtifacts: 
   - beam.runners.flink:1.10:2.28.0-SNAPSHOT@jar
```

As it is transitively being declared, but we do not need it here. Thats why it is important to declare it permitUnsedDeclared.